### ISSUE TITLE
Build failure: klipper-firmware

### ISSUE THREAD

**Issue by arachnist** [NixOS/nixpkgs#263370]
### Steps To Reproduce

1. `klipper-firmware` builds targeting `lpc176x` boards/`lpc1769` mcu (possibly other arm MCUs) fails with `gcc-arm-embedded-12` (which is the default now).
### Build log

[log](https://gist.github.com/arachnist/b9f4f569db9b0d75147370b5bf433a50)

### Additional context

Building with gcc 11, eg. as below, works.

```console
$ nix-shell -p '(klipper-firmware.override { firmwareConfig = ./. + "/klipper-smoothie.cfg"; gcc-arm-embedded = gcc-arm-embedded-11; })'
```

### Notify maintainers

@vtuan10

### Metadata

Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
❯ nix-shell -p nix-info --run "nix-info -m"
 - system: `"aarch64-linux"`
 - host os: `Linux 6.5.7, NixOS, 23.11 (Tapir), 23.11.20231019.7c9cc5a`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.18.1`
 - channels(root): `"agenix, nixos"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

**Comment by vtuan10** [comment]
https://github.com/Klipper3d/klipper/pull/6331

Seems like there exist a draft PR for this issue which is in progress. Is it alright to pin it to `gcc-arm-embedded-11` for now?

**Comment by arachnist** [comment]
I don't have at hand the printer I want to use klipper with to test if it actually works, but with `gcc-arm-embedded-11` it at least builds.

**Comment by arachnist** [comment]
Had a chance to test the firmware binary built this way, and it works.

**Comment by fpletz** [comment]
Fixed by  #263813.
